ui/vty-lib/src/Ariadne/UI/Vty/Widget/Dialog/ConfirmRemove.hs

```diff
@@ -0,0 +1,92 @@
+module Ariadne.UI.Vty.Widget.Dialog.ConfirmRemove
+    ( initConfirmRemoveWidget
+    ) where
+
+import Control.Lens (makeLensesWith, (.=))
+
+import qualified Brick as B
+import qualified Brick.Widgets.Center as B
+
+import Ariadne.UI.Vty.Face
+import Ariadne.UI.Vty.Keyboard
+import Ariadne.UI.Vty.Widget
+import Ariadne.UI.Vty.Widget.Dialog.Utils
+import Ariadne.Util
+
+data ConfirmRemoveWidgetState = ConfirmRemoveWidgetState
+    { confirmRemoveWidgetUiFace    :: !UiFace
+    , confirmRemoveWidgetResultVar :: !(Maybe (MVar Bool))
+    , confirmRemoveWidgetDelItem   :: !(Maybe (UiDeletingItem))
+    }
+
+makeLensesWith postfixLFields ''ConfirmRemoveWidgetState
+
+initConfirmRemoveWidget :: UiFace -> Widget p
+initConfirmRemoveWidget uiFace = initWidget $ do
+    setWidgetDrawWithFocus drawConfirmRemoveWidget
+    setWidgetHandleKey handleConfirmRemoveWidgetKey
+    setWidgetHandleEvent handleConfirmRemoveWidgetEvent
+    setWidgetState ConfirmRemoveWidgetState
+        { confirmRemoveWidgetUiFace    = uiFace
+        , confirmRemoveWidgetResultVar = Nothing
+        , confirmRemoveWidgetDelItem   = Nothing
+        }
+
+    addDialogButton WidgetNameConfirmRemoveCancel "Cancel" performCancel
+    addDialogButton WidgetNameConfirmRemoveContinue "Continue" performContinue
+
+    setWidgetFocusList
+        [ WidgetNameSelf
+        , WidgetNameConfirmRemoveContinue
+        , WidgetNameConfirmRemoveCancel
+        ]
+
+drawConfirmRemoveWidget
+    :: WidgetName
+    -> ConfirmRemoveWidgetState
+    -> WidgetDrawM ConfirmRemoveWidgetState p WidgetDrawing
+drawConfirmRemoveWidget focus ConfirmRemoveWidgetState{..} =
+    case confirmRemoveWidgetResultVar of
+        Nothing -> return $ singleDrawing B.emptyWidget
+        Just _  -> drawInsideDialog "Confirm ConfirmRemove" focus
```

Comment:
Do you plan to add retyping to tui?

Reply:
Yes, I will look putting that back. The fact is that the backend launches the event (that starts the dialog) with the data, and it doesn't have the name of the wallet/account.
This is why I removed it temporarily.

Reply:
Sorry I misread the question (I will leave the answer above as an answer to why retyping is not on GUI anymore). Anyway, I don't know, I suppose I will if I put it back on GUI.

Reply:
You may want to take a look at what I did for new address command in gui. There I pass back account index from the original command, not from result. See resultToUI in glue.
If you get at least wallet index or root id back, you can look up its name in UI state, I suppose.